io/disk/vfc-tests/vfc_dlpar.py

```diff
@@ -0,0 +1,367 @@
+from avocado import skipIf, skipUnless
+
+IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()
+IS_KVM_GUEST = 'qemu' in open('/proc/cpuinfo', 'r').read()
+
+
+class CommandFailed(Exception):
+    '''
+    Defines the exception called when a
+    command fails
+    '''
+
+    def __init__(self, command, output, exitcode):
+        Exception.__init__(self, command, output, exitcode)
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class VirtualFC(Test):
+    '''
+    Removing and Adding and Fibre Chanel Virtualized devices from the HMC
+    '''
+    @skipUnless("ppc" in distro.detect().arch,
+                "supported only on Power platform")
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST,
+            "This test is not supported on KVM guest or PowerNV platform")
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        self.install_packages()
+        self.rsct_service_start()
+        self.hmc_ip = self.params.get("hmc_ip", default=None)
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.count = self.params.get("count", default=1)
+        self.opp_sleep_time = 150
+        self.lpar = self.get_partition_name("Partition Name")
+        if not self.lpar:
+            self.cancel("LPAR Name not got from lparstat command")
+        self.login(self.hmc_ip, self.hmc_username, self.hmc_pwd)
+        cmd = 'lssyscfg -r sys  -F name'
+        output = self.run_command(cmd)
+        self.server = ''
+        for line in output:
+            if line in "ltcfleet2-lp10-Naresh":
+                self.server = line
+        if not self.server:
+            self.cancel("Managed System not got")
+        self.dic_list = []
+        cmd = 'lshwres -r virtualio --rsubtype fc --level lpar -m %s \
+               --filter "lpar_names=%s"' % (self.server, self.lpar)
+        for line in self.run_command(cmd):
+            self.vfc_dic = {}
+            for i in line.split(","):
+                if i.split("=")[0] == "slot_num":
+                    self.vfc_dic["c_slot"] = i.split("=")[-1]
+                elif i.split("=")[0] == "remote_slot_num":
+                    self.vfc_dic["r_slot"] = i.split("=")[-1]
+                elif i.split("=")[0] == "remote_lpar_name":
+                    self.vfc_dic["r_lpar"] = i.split("=")[-1]
+            self.vfc_dic["wwpn"] = self.get_wwpn(self.vfc_dic["c_slot"])
+            self.vfc_dic["drc"] = self.get_drc_name(self.vfc_dic["c_slot"])
+            self.vfc_dic["paths"] = self.get_paths(self.vfc_dic["drc"])
+            self.dic_list.append(self.vfc_dic)
+            self.log.info("individual dictionaries: %s " % self.vfc_dic)
```

Comment:
Is this needed, since we are printing the complete list as well ?

Reply:
can be removed. for quick debug I have added that line.